### PR TITLE
CHEF-3217 - Patch with file path 

### PR DESCRIPTION
**Pull request by shanmugapriya-tr**
### :nut_and_bolt: Description: What code changed, and why?
Introduced two variables in Command input : SourceConfig and DestinationConfig. 
SourceConfig holds the entire config file path provided by the user. DestinationConfig holds the timestamp formatted file name that will be copied on the nodes. 

### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-3217

### :+1: Definition of Done
User should be able to patch config present in any folder path

### :athletic_shoe: How to Build and Test the Change
Checkout the branch
rebuild components/automate-cli

use the package to test changes

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
Screenshot 2023-06-01 at 2 38 57 PM
Screenshot 2023-06-01 at 2 39 17 PM
Screenshot 2023-06-01 at 2 38 08 PM
Screenshot 2023-06-01 at 2 36 30 PM

Screenshot 2023-06-01 at 2 35 44 PM
Screenshot 2023-06-01 at 2 35 35 PM
Screenshot 2023-06-01 at 2 35 23 PM

Screenshot 2023-06-01 at 2 34 16 PM
Screenshot 2023-06-01 at 2 33 40 PM

### CODE REVIEW

File: components/automate-cli/cmd/chef-automate/config.go [hunk 1]
```diff
@@ -368,9 +375,11 @@ func runPatchCommand(cmd *cobra.Command, args []string) error {
 				InputFiles:               []string{configFile},
 				ErrorCheckEnableInOutput: true,
 				NodeType:                 configCmdFlags.frontend,
+				SourceConfig:             configFile,
+				DestinationConfig:        "frontend" + "_" + timestamp + "_" + configFileRenamed,
 			},
 		}
-		automateCmd := fmt.Sprintf(FRONTEND_COMMAND, PATCH, "automate"+"_"+timestamp+"_"+configFile, dateFormat)
+		automateCmd := fmt.Sprintf(FRONTEND_COMMAND, PATCH, "automate"+"_"+timestamp+"_"+configFileRenamed, dateFormat)
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/config.go:382]: Same in this case

File: components/automate-cli/cmd/chef-automate/config.go [hunk 2]
```diff
@@ -368,9 +375,11 @@ func runPatchCommand(cmd *cobra.Command, args []string) error {
 				InputFiles:               []string{configFile},
 				ErrorCheckEnableInOutput: true,
 				NodeType:                 configCmdFlags.frontend,
+				SourceConfig:             configFile,
+				DestinationConfig:        "frontend" + "_" + timestamp + "_" + configFileRenamed,
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/config.go:379]:
```suggestion
				DestinationConfig:        destinationConfigFileName,
```

File: components/automate-cli/cmd/chef-automate/config.go (hunk 1 above)
Comment (kalroy) [reply components/automate-cli/cmd/chef-automate/config.go:382]:
```suggestion
		destinationAutomateConfigFileName := fmt.Sprintf("automate_%s_%s",timestamp, sourceConfigFileName)
		frontendCmd := fmt.Sprintf(FRONTEND_COMMAND, PATCH, destinationAutomateConfigFileName, dateFormat)
```

File: components/automate-cli/cmd/chef-automate/config.go [hunk 3]
```diff
@@ -381,10 +390,12 @@ func runPatchCommand(cmd *cobra.Command, args []string) error {
 				InputFiles:               []string{configFile},
 				ErrorCheckEnableInOutput: true,
 				NodeType:                 configCmdFlags.automate,
+				SourceConfig:             configFile,
+				DestinationConfig:        "automate" + "_" + timestamp + "_" + configFileRenamed,
```
Comment (kalroy) [inline components/automate-cli/cmd/chef-automate/config.go:394]:
```suggestion
				DestinationConfig:        destinationAutomateConfigFileName,
```

**Review by kalroy — CHANGES_REQUESTED** [review]

**Review by YashviJain01 — APPROVED** [review]

File: components/automate-cli/cmd/chef-automate/cmdUtils.go [hunk 4]
```diff
@@ -140,14 +142,19 @@ func (c *remoteCmdExecutor) executeCmdOnGivenNodes(input *CmdInputs, nodeIps []s
...
 	timeout := input.WaitTimeout
 	inputFileToOutputFileMap := map[string]string{}
-	for _, file := range inputFiles {
-		destinationFile := remoteService + "_" + timestamp + "_" + file
-		if strings.Contains(file, "/") {
-			filePath := strings.Split(file, "/")
-			lastFileidx := len(filePath) - 1
-			destinationFile = filePath[lastFileidx]
+
+	if input.SourceConfig != "" && input.DestinationConfig != "" {
```
Comment (vivek-yadav) [inline components/automate-cli/cmd/chef-automate/cmdUtils.go:146]: This approach is different, we can pass the prefix of destination files, and then use that in all locations.

Please connect with me, I can tell you more about this.

**Pull request closed without merging**